docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructureImpl.java

```diff
@@ -427,6 +421,32 @@ public void stop() {
         deleteLocation();
     }
 
+    @Override
+    public void postStart(Collection<? extends Location> locations) {
+        super.postStart(locations);
+
+
+        if(config().get(DOCKER_SHOULD_START_REGISTRY)){
+            DockerHost firstEntity = (DockerHost) sensors().get(DOCKER_HOST_CLUSTER).sensors().get(DynamicCluster.FIRST);
+
+
+            EntitySpec<DockerRegistry> spec = EntitySpec.create(DockerRegistry.class)
+                    .configure(DockerRegistry.DOCKER_HOST, firstEntity);
+
+            //mount volume with images stored on it
+            DockerRegistry dockerRegistry = addChild(spec);
+            Entities.manage(dockerRegistry);
+
+            LOG.debug("Starting a new Docker Registry with spec {}", spec);
+            dockerRegistry.start(ImmutableList.of(firstEntity.getDynamicLocation()));
+
```

Comment:
Better to use `Entities#Start(Coillection)` for this, which will manage the entity and call start in a new `Task`.